xiangqi_lite/src/pages/Game/BoardPage.jsx

```diff
@@ -0,0 +1,639 @@
+            
+            
+            setCurrentIndex(history.length - 1);
+            
+           
+            handleParseFENInput(history[history.length - 1].fen);
+            
+            
+            setTimeout(() => {
+                setShowMoveRestrictionBanner(false);
+            }, 5000);
+        }
+    }, [currentIndex, history, setCurrentIndex, handleParseFENInput]);
+    
+    
+
+    const addUser = (newUsers) => {
+        setUsers(prevUsers => {
+          const usersData = { ...prevUsers }; 
+          if (Array.isArray(newUsers)) {
+            newUsers.forEach((user) => {
+              usersData[user.id] = user; 
+            });
+          } else if (newUsers.id) {
+            usersData[newUsers.id] = newUsers; 
+          }
+          return usersData; 
+        });
+      };
+    
+    const handleTimerExpire = () => {
+        console.log(`${usernameRef.current} timer expired!`);
+            
+            if (wsManagerRef.current && wsManagerRef.current.isConnected) {
+                const message = JSON.stringify({
+                    type: 'game.end',
+                    game_id: gameIdRef.current,
+                    losing_player:usernameRef.current,
+                });
+                wsManagerRef.current.sendMessage(message);
+            }
+        } 
+
+    const updateChatMessages = (chatMessage) => {
+        setChatMessages(prevMessages => [...prevMessages, chatMessage]);
+    };
+    const handleGameGetSuccess = (gameData) => {
+        setFENOutput(gameData.fen);
+        handleParseFENInput(gameData.fen);
+        setCurrentTurn(gameData.turn);
+        console.log("Current turn on game start", latestCurrentTurn.current);
+        gameIdRef.current = gameData.game_id;  
+    
+        if (usernameRef.current === gameData.black_player) {
+            handleFlipBoard(); 
+        }
+    
+        setRedTimeRemaining(gameData.red_time_remaining);
+        setBlackTimeRemaining(gameData.black_time_remaining);
+
+        setHistory([]);
+    };
+    
+    const handleUserListData = (userData) => {
+        if (userData.length > 0) {
+            setRedPlayer(userData[0]);
+            setBlackPlayer(userData[1]);
+        }
+    };
+
+    const gameover = () => {
+        checkGameOver(latestSqr.current,latestCurrentTurn.current,findAvailableSqr,setGameOver,handleTimerExpire)
+    };
+    
+    const {
+        redTimeRemaining,
+        setRedTimeRemaining,
+        blackTimeRemaining,
+        setBlackTimeRemaining,
+        setIsRedTimerRunning,
+        setIsBlackTimerRunning,
+        
+    } = useGameTimer(300, 300,handleTimerExpire,isPlayerAllowedToMove);
+
+    const {
+        redMoveTimeRemaining,
+        setRedMoveTimeRemaining, 
+        blackMoveTimeRemaining, 
+        setBlackMoveTimeRemaining, 
+        setIsRedMoveTimerRunning, 
+        setIsBlackMoveTimerRunning
+    } = useMoveTimer(60, 60, handleTimerExpire,isPlayerAllowedToMove);
```

Comment:
Such states from hooks should be added on top and then functions / callbacks / events